src/models/flinkStatement.ts

```diff
@@ -62,10 +136,53 @@ export class FlinkStatementTreeItem extends TreeItem {
     this.contextValue = `${resource.connectionType.toLowerCase()}-flink-statement`;
 
     // user-facing properties
-    this.iconPath = createFlinkStatementIcon(resource.status);
-    this.description = resource.status;
 
-    // TODO: add tooltip
+    // (Sometimes sqlKindDisplay is undefined, e.g. for failed statements)
+    this.description = resource.sqlKindDisplay
+      ? `${resource.phase} ${resource.sqlKindDisplay}`
+      : resource.phase;
+    this.iconPath = this.getThemeIcon();
+
+    this.tooltip = CustomMarkdownString.resourceTooltip(
+      "Flink Statement",
+      this.iconPath.id as IconNames,
+      resource.ccloudUrl,
+      [
+        ["Kind", resource.sqlKindDisplay],
+        ["Status", resource.phase],
+        ["Created At", resource.createdAt?.toLocaleString()],
+        ["Updated At", resource.updatedAt?.toLocaleString()],
+        ["Environment", resource.environmentId],
+        ["Compute Pool", resource.computePoolId],
+        ["Detail", resource.status.detail],
+      ],
+    );
+  }
+
+  /**
+   * Determine icon + color based on the `phase` of the statement.
+   */
+  getThemeIcon(): ThemeIcon {
```

Comment:
Nit: only downside with this being a tree item method is we won't be able to use it as easily if we ever need a statement quickpick